banking/custom/purchase_invoice.py

```diff
@@ -28,35 +28,36 @@ def set_missing_values(source, target):
 				target.iban = bank_account.get("iban")
 				target.bank = bank_account.get("bank")
 
-	def process_payment(source, target, source_parent):
-		target.recipient = source_parent.supplier_name
-		target.purpose = source_parent.bill_no
-		target.currency = source_parent.currency
-		target.eref = target.reference_name
-
-		if source_parent.supplier_bank_account:
-			# Prefer the Supplier Bank Account set on the Purchase Invoice
-			bank_account = frappe.db.get_value(
-				"Bank Account",
-				source_parent.supplier_bank_account,
-				["iban", "bank"],
-				as_dict=True,
-			)
-		else:
-			# Fallback to the (default) Bank Account linked to the Supplier
-			bank_account = frappe.db.get_value(
-				"Bank Account",
-				{"party_type": "Supplier", "party": source_parent.supplier, "disabled": 0},
-				["iban", "bank"],
-				order_by="is_default DESC",
-				as_dict=True,
-			)
-
+	def process_payment(source, target, purchase_invoice):
+		is_employee_advance_paid = (
+			hasattr(purchase_invoice, "business_trip")
+			and purchase_invoice.business_trip
+			and hasattr(purchase_invoice, "business_trip_employee")
+			and purchase_invoice.business_trip_employee
+			and hasattr(purchase_invoice, "pay_to_employee")
+			and purchase_invoice.pay_to_employee
+		)
```

Comment:
```suggestion
		pay_to_employee = all((
			hasattr(purchase_invoice, "business_trip") and purchase_invoice.business_trip,
			hasattr(purchase_invoice, "business_trip_employee") and purchase_invoice.business_trip_employee,
			hasattr(purchase_invoice, "pay_to_employee") and purchase_invoice.pay_to_employee,
		))
```

Das ist sicher Geschmacksache, aber so sieht man meiner Meinung nach besser, dass es drei Voraussetzungen gibt, die alle erfüllt sein müssen.

Und die Variable könnte man auch noch umbenennen. :D